File: 素数筛/sim_opti_algo.py
"""素数筛：朴素算法，时间复杂度为O(sqrt(n)*n)"""

num=100
primes=[2]

import math
count=1
for i in range(3,num):
    if all(i%j!=0 for j in range (2,int(math.sqrt(i)+1))):
        primes.append(i)
        count+=1

for i in range(0,count):
    print(f"{primes[i]:5d}",end="")
    if (i+1)%5==0:
        print()

File: algo/素数筛/sim_opti_algo.c
/*素数筛：朴素算法，时间复杂度为O(n)*/
#include <stdio.h>
#include <math.h>
#define num 100

int prime_arr[num];

int prime(int n)
{

    prime_arr[0] = 2;

    int i, j, k = 1;
    _Bool flag;

    for (i = 3; i < n; i++)
    {
        for (j = 2; j <= sqrt(i); j++)
        {
            flag = 1;
            if (i % j == 0)
            {
                flag = 0;
                break;
            }
        }
        if (flag == 1)
        {
            prime_arr[k] = i;
            k++;
        }
    }
    return k;
}

int main()
{
    int i;
    int k = prime(num);

    for (i = 0; i < k; i++)
    {
        printf("%5d", prime_arr[i]);
        if ((1 + i) % 5 == 0)
            printf("\n");
    }
    return 0;
}